packages/language-kit/src/type.ts

```diff
@@ -0,0 +1,11 @@
+import { z } from 'zod';
+
+import { languages } from './const';
+
+export type LanguageTag = keyof typeof languages;
+
+export const isLanguageTag = (tag: string): tag is LanguageTag => tag in languages;
+
+export const languageTagGuard: z.ZodType<LanguageTag> = z
+  .any()
```

Comment:
can we use `z.enum`?

Reply:
Cannot. The parameter type of `z.enum()` should be like

<img width="905" alt="image" src="https://user-images.githubusercontent.com/10594507/192194691-b8efe5ba-a3a8-49d7-98d5-dd65af36975c.png">

We have to declare a read-only literal array `languageTags` (separated from `languages`) to match the type for `z.enum(…)`.

It's unnecessary and duplicated with `languages`.

And you cannot use `languages` for `z.enum(…)` as below (I have tried many ways).

<img width="1312" alt="image" src="https://user-images.githubusercontent.com/10594507/192194665-b854104b-c74a-4905-ae93-3a5fc8034f75.png">